src/commands/periscope/helpers/periscopehelper.ts

```diff
@@ -0,0 +1,265 @@
+    try {
+        // Get daignostic setting via diagnostic monitor
+        const diagnosticMonitor = new amon.MonitorManagementClient(cluster.root.credentials, cluster.root.subscriptionId);
+        const diagnosticSettings = await diagnosticMonitor.diagnosticSettingsOperations.list(cluster.id!);
+
+        return diagnosticSettings;
+    } catch (e) {
+        vscode.window.showErrorMessage(`Error fetching cluster diagnostic monitor: ${e}`);
+        return undefined;
+    }
+}
+
+export async function chooseStorageAccount(
+    diagnosticSettings: amon.MonitorManagementModels.DiagnosticSettingsResourceCollection | undefined,
+): Promise<string | undefined> {
+    /*
+        Check the diagnostic setting is 1 or more than 1:
+          1. For the scenario of 1 storage account in diagnostic settings - Pick the storageId resource and get SAS.
+          2. For the scenario for more than 1 then show VsCode quickPick to select and get SAS of selected.
+    */
+
+    if (diagnosticSettings && diagnosticSettings.value!.length > 1) {
+        const storageAccountNameToStorageIdArray: { id: string; label: string; }[] = [];
+
+        diagnosticSettings.value?.forEach((item) => {
+            if (item.storageAccountId) {
+                const { name } = parseResource(item.storageAccountId!);
+                if (!name) {
+                    vscode.window.showInformationMessage(`Storage Id is malformed: ${item.storageAccountId}`);
+                    return;
+                }
+                storageAccountNameToStorageIdArray.push({ id: item.storageAccountId, label: name });
+            }
+        });
+
+        // accounts is now an array of {id, name}
+        const accountQuickPicks = storageAccountNameToStorageIdArray;
+
+        // Create quick pick for more than 1 storage account scenario.
+        const selectedQuickPick = await vscode.window.showQuickPick(
+            accountQuickPicks,
+            {
+                placeHolder: "Select storage account for Periscope deployment:",
+                ignoreFocusOut: true,
+                onDidSelectItem: (item: string) => { return item.toString(); }
+            });
+
+        if (selectedQuickPick?.label) {
+            return selectedQuickPick?.id;
+        }
+
+    } else if (diagnosticSettings && diagnosticSettings.value!.length === 1) {
+        // In case of only one storage account associated, use the one (1) as default storage account and no UI will be displayed.
+        const selectedStorageAccount = diagnosticSettings.value![0].storageAccountId!;
+        return selectedStorageAccount;
+    }
+
+    return undefined;
+}
+
+export async function getStorageInfo(
+    cluster: AksClusterTreeItem,
+    diagnosticStorageAccountId: string
+): Promise<PeriscopeStorage | undefined> {
+    try {
+        const { resourceGroupName, name: accountName } = parseResource(diagnosticStorageAccountId!);
```

Comment:
You shouldn't need the dammit operator - `diagnosticStorageAccountId` is already non-nullable.